crates/services/sync/src/import.rs

```diff
@@ -163,41 +161,22 @@ where
             consensus,
             ..
         } = &self;
-        // Request up to `max_get_header_requests` headers from the network.
-        get_header_range_buffered(range.clone(), params, p2p.clone())
+
+        let headers_res = get_header_range_single_req(range.clone(), p2p.clone()).await;
+        let unchecked_headers = match headers_res {
+            Ok(headers) => headers,
+            Err(e) => return (0, Err(e)),
+        };
+        let headers = check_headers_match_range(&unchecked_headers, range.clone());
```

Comment:
I think it is better to move it inside of the `map` section below

Reply:
Is there a reason you'd prefer this? I just gave this a try and it got kinda messy trying to get the future types to match up.